.github/workflows/publish.yaml

```diff
@@ -0,0 +1,34 @@
+name: Publish
+
+on:
+  push:
+    tags:
+      - '0.*'
+      - '1.*'
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      attestations: write
+      contents: read
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+
+      - name: Build dist packages
+        run: uv build
+
+      - name: Attest build provenance
+        uses: actions/attest-build-provenance@v2.2.3
+        with:
+          subject-path: 'dist/*'
+
+      - name: Publish
+        uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
It's not needed right now since there aren't (as far as I know) any secrets, and maybe there never will be. But it's probably worth using a separate environment here, which protects against mistakes being made in the future. Particularly given the recent actions attack that hit a lot of people, it seems worth doing the small amount of extra work in advance.

Reply:
Good call -- I've added a `publish-pypi` environment in GitHub.